### DISCUSSION TITLE
Increase/decrease weight hotkeys

### DISCUSSION THREAD

**Discussion by missionfloyd** [comfyanonymous/ComfyUI#485] [Show and tell]
Apparently this can't be merged, so I'll just leave it here if anyone wants it.

Edit weights with `Ctrl + up/down`, just like in Auto1111

https://github.com/missionfloyd/ComfyUI-editattention

**Comment by omar92** [comment]
i think we can use diffrent buttons or try change something to make it legal 

maybe if you click button on a word it open a popup then you use it to write the weight

**Reply by comfyanonymous** [reply]
if you want to "make it legal" you just need to write new code that does the same thing without copying anything from the A1111 one.

We can't copy anything from the A1111 codebase because it's licensed AGPL which isn't compatible with the GPLv3+ we use because AGPL is way more restrictive. Even if it was licensed in a compatible license the A1111 repo didn't have a license for a long time and the maintainer arbitrarily put one without asking permission from prior contributors first which makes it invalid and the code a legal mess.

The only way I'm accepting code from A1111 is if the person who submits it is the one that originally wrote the code.

**Reply by omar92** [reply]
i can do this , i thought you can't do the same thing , thanks for clarifying this

**Reply by WASasquatch** [reply]
> if you want to "make it legal" you just need to write new code that does the same thing without copying anything from the A1111 one.
> 
> We can't copy anything from the A1111 codebase because it's licensed AGPL which isn't compatible with the GPLv3+ we use because AGPL is way more restrictive. Even if it was licensed in a compatible license the A1111 repo didn't have a license for a long time and the maintainer arbitrarily put one without asking permission from prior contributors first which makes it invalid and the code a legal mess.
> 
> The only way I'm accepting code from A1111 is if the person who submits it is the one that originally wrote the code.

Reverse engineering and writing your own version of something is perfectly legal in this instance. This isn't a patent. Just don't copypasta. 

FYI reverse engineering is a perfectly acceptable means of gathering information and ideas.
